File: benchmarks/correctness/real_world/python/gff_features.py
import json

total = 0
by_type = {}
with open("real_data/ecoli_annotation.gff") as f:
    for line in f:
        if line.startswith("#") or not line.strip():
            continue
        total += 1
        parts = line.strip().split("\t")
        if len(parts) >= 3:
            ftype = parts[2]
            by_type[ftype] = by_type.get(ftype, 0) + 1

print(json.dumps({"total_features": total, "by_type": by_type}))

File: benchmarks/correctness/real_world/r/gff_features.R
library(jsonlite)
lines <- readLines("real_data/ecoli_annotation.gff")
lines <- lines[!grepl("^#", lines) & nchar(trimws(lines)) > 0]
total <- length(lines)
types <- sapply(lines, function(l) strsplit(l, "\t")[[1]][3])
by_type <- as.list(table(types))
by_type <- lapply(by_type, as.integer)
cat(toJSON(list(total_features = total, by_type = by_type), auto_unbox = TRUE))
cat("\n")
